### ISSUE TITLE
Initial synchronization bug of virtual  locationregistry

### ISSUE THREAD

**Issue by pholthau** [openbase/bco.registry#39]
```18:03:59.619 [Thread-2] ERROR o.o.b.r.l.r.LocationRegistryRemote =====================================
18:03:59.619 [Thread-2] ERROR o.o.b.r.l.r.LocationRegistryRemote ═╦═⚀ Could not resolve UnitConfigId[a69b3e90-f0be-4433-9a40-583b438a3b93] by device registry!
18:03:59.619 [Thread-2] ERROR o.o.b.r.l.r.LocationRegistryRemote  ╚══╦═⚀ Entry[a69b3e90-f0be-4433-9a40-583b438a3b93]] is not available!
18:03:59.619 [Thread-2] ERROR o.o.b.r.l.r.LocationRegistryRemote     ╚════⚀ SynchronizedRemoteRegistry[UnitRegistryRemote[scope:/registry/unit/]][dal_unit_config, user_unit_config, authorization_group_unit_config, device_unit_config, unit_group_unit_config, location_unit_config, connection_unit_config, agent_unit_config, scene_unit_config, app_unit_config] is empty!
18:03:59.619 [Thread-2] ERROR o.o.b.r.l.r.LocationRegistryRemote =====================================
18:03:59.619 [Thread-2] ERROR o.o.b.r.l.r.LocationRegistryRemote Could not resolve UnitConfigId[a69b3e90-f0be-4433-9a40-583b438a3b93] by device registry!
org.openbase.jul.exception.CouldNotPerformException: Could not resolve UnitConfigId[a69b3e90-f0be-4433-9a40-583b438a3b93] by device registry!
	at org.openbase.bco.registry.location.remote.LocationRegistryRemote.getUnitConfigsByLocation(LocationRegistryRemote.java:289)
	at org.openbase.bco.registry.location.remote.LocationRegistryRemote.getUnitConfigsByLocationLabel(LocationRegistryRemote.java:305)
	at de.citec.csra.aa.light.LightExecutor.units(LightExecutor.java:61)
	at de.citec.csra.aa.light.LightExecutor.exec(LightExecutor.java:80)
	at de.citec.csra.aa.light.LightExecutor.off(LightExecutor.java:75)
	at de.citec.csra.aa.ResultHandler.handle(ResultHandler.java:50)
	at de.citec.csra.aa.RSBResultHandler.internalNotify(RSBResultHandler.java:31)
	at rsb.eventprocessing.SingleThreadEventReceivingStrategy$DispatchThread.run(SingleThreadEventReceivingStrategy.java:98)
Caused by: org.openbase.jul.exception.NotAvailableException: Entry[a69b3e90-f0be-4433-9a40-583b438a3b93]] is not available!
	at org.openbase.jul.storage.registry.AbstractRegistry.get(AbstractRegistry.java:305)
	at org.openbase.jul.storage.registry.RemoteRegistry.getMessage(RemoteRegistry.java:78)
	at org.openbase.bco.registry.unit.remote.UnitRegistryRemote.getUnitConfigById(UnitRegistryRemote.java:229)
	at org.openbase.bco.registry.location.remote.LocationRegistryRemote.getUnitConfigsByLocation(LocationRegistryRemote.java:284)
	... 7 common frames omitted
Caused by: org.openbase.jul.exception.InvalidStateException: SynchronizedRemoteRegistry[UnitRegistryRemote[scope:/registry/unit/]][dal_unit_config, user_unit_config, authorization_group_unit_config, device_unit_config, unit_group_unit_config, location_unit_config, connection_unit_config, agent_unit_config, scene_unit_config, app_unit_config] is empty!
	... 11 common frames omitted
```

**Comment by DivineThreepwood** [comment]
Fix should be backported to latest-stable branch v1.3.X if possible.

**Comment by DivineThreepwood** [comment]
There seems to be two issues:
* The waitForData Future is nearly returning immediately while the internal registry is still empty.
* The location registry is resolving devices or the logger message is just wrong.
    * ```ERROR o.o.b.r.l.r.LocationRegistryRemote Could not resolve UnitConfigId[a69b3e90-f0be-4433-9a40-583b438a3b93] by device registry!```

**Comment by pLeminoq** [comment]
The logger message was wrong and waitForData for RegistryRemote which listen to VirtualRegistries did not work correctly. Should be fixed by commits:

* cb00d40860aeb6fc6109fd2586d520c9082d1b43
* 31edfb4db2b400e40df9dbc93b29c13a10de188c

Backporting of those commits still has to be done.

**Comment by pLeminoq** [comment]
Additional commit which fixed unit tests:

11cdef9c13afa0361b5a804416f46e5e2965703e

**Comment by pLeminoq** [comment]
Released version 1.3.3
